app/models/guest_device.rb

```diff
@@ -10,4 +11,7 @@ class GuestDevice < ApplicationRecord
 
   has_one :network, :foreign_key => "device_id", :dependent => :destroy, :inverse_of => :guest_device
   has_many :miq_scsi_targets, :dependent => :destroy
+
+  has_many :firmwares, :dependent => :destroy
+  has_many :child_devices, -> { where(:parent_device_id => ids) }, :foreign_key => "parent_device_id", :class_name => "GuestDevice", :dependent => :destroy
```

Comment:
@blomquisg @juliancheal I updated the PR with the suggested changes. The associated changes in the provider were merged, so they're in master which is what I'm testing with.